mailet/base/src/main/java/org/apache/mailet/base/MailetUtil.java

```diff
@@ -109,4 +122,90 @@ public static boolean getInitParameter(MailetConfig config, String name, boolean
         }
         return result;
     }
+
+    public static boolean canSendAutomaticResponse(Mail mail) throws MessagingException {
+        return !isMailingList(mail) &&
+            ! isAutoSubmitted(mail) &&
+            ! isMdnSentAutomatically(mail);
+    }
+
+    public static boolean isMailingList(Mail mail) throws MessagingException {
+        return mail.getSender().getLocalPart().startsWith("owner-")
+            || mail.getSender().getLocalPart().endsWith("-request")
+            || mail.getSender().getLocalPart().equalsIgnoreCase("MAILER-DAEMON")
+            || mail.getSender().getLocalPart().equalsIgnoreCase("LISTSERV")
+            || mail.getSender().getLocalPart().equalsIgnoreCase("majordomo")
+            || mail.getMessage()
+                .getMatchingHeaders(new String[]{"List-Help",
+                    "List-Subscribe",
+                    "List-Unsubscribe",
+                    "List-Owner",
+                    "List-Post",
+                    "List-Id",
+                    "List-Archive"})
+                .hasMoreElements();
+    }
+
+    public static boolean isAutoSubmitted(Mail mail) throws MessagingException {
+        String[] headers = mail.getMessage().getHeader("Auto-Submitted");
+        if (headers != null && headers.length > 0) {
```

Comment:
You meant I can never receive a 0 sized array ? I have no such trust on getHeader....